src/ray/raylet/node_manager.cc

```diff
@@ -2202,8 +2198,8 @@ void NodeManager::HandleGetNodeStats(const rpc::GetNodeStatsRequest &node_stats_
   object_manager_.FillObjectStoreStats(reply);
   // Ensure we never report an empty set of metrics.
   if (!recorded_metrics_) {
-    RecordMetrics();
-    RAY_CHECK(recorded_metrics_);
+    DebugString();
```

Comment:
A little weird. Here you don't need to get the debug string.

Reply:
Maybe the debug function could be like:
```
std::string RecordStats(bool return_debug_string=true, bool record_metrics=true);
```
?

Reply:
I think RecordStats is also the wrong name what about

```
std::string DebugString(bool record=true) const;
```
? 

Btw, why do we not need empty metrics here? Can we just remove this logic?

Reply:
It is ok If you always need a debug string.
I don't know. I'm not familiar with this part.

Reply:
Hmm I think we need some clean interface here. I will discuss with @scv119 and get back here.